zebra-network/src/isolated/tor.rs

```diff
@@ -45,6 +45,41 @@ pub async fn connect_isolated_tor(
     hostname: String,
     user_agent: String,
 ) -> Result<BoxService<Request, Response, BoxError>, BoxError> {
+    let tor_stream = new_tor_stream(hostname).await?;
+
+    // Calling connect_isolated_tor_with_inbound causes lifetime issues
+    connect_isolated(network, tor_stream, user_agent).await
+}
+
+/// Creates an isolated Zcash peer connection to `hostname` via Tor.
+/// This method is for testing purposes only.
+///
+/// See [`connect_isolated_with_inbound`] and [`connect_isolated_tor`] for details.
+///
+/// # Privacy
+///
+/// This method can make the isolated connection send different responses to peers,
+/// which makes it stand out from other isolated connections from other peers.
+pub async fn connect_isolated_tor_with_inbound<InboundService>(
```

Comment:
Question: Would it make sense to make this function be only available for test builds?

```suggestion
#[cfg(any(test, feature = "proptest-impl"))]
pub async fn connect_isolated_tor_with_inbound<InboundService>(
```

Reply:
No, I'd rather call it from the non-test method, so that their behaviour always remains consistent.

Added a TODO in e9125041b.
(This TODO doesn't need a ticket - we'll fix the Tor API during or after the `arti` upgrade.)